rixed told me (viric) it's the patch to apply for mesa.
I adapted his patch for mesa 7.5 to this 7.10

diff --git a/src/mesa/drivers/dri/sis/sis_context.h b/src/mesa/drivers/dri/sis/sis_context.h
index a82a659..d8d8d9a 100644
--- a/src/mesa/drivers/dri/sis/sis_context.h
+++ b/src/mesa/drivers/dri/sis/sis_context.h
@@ -404,6 +404,8 @@ struct sis_context
 #define MMIO_WMB()	__asm __volatile("" : : : "memory")
 #elif defined(__ia64__)
 #define MMIO_WMB()	__asm __volatile("mf" : : : "memory")
+#elif defined(__mips__)
+#define MMIO_WMB()     __asm __volatile("" : : : "memory")
 #else
 #error platform needs WMB
 #endif
